core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -503,8 +503,9 @@ private void updateCacheStream(long blockId) {
     }
     Preconditions.checkNotNull(mCurrentBlockInStream, "mCurrentBlockInStream");
 
-    // do not create cache stream when the block is in local worker
-    if (!mShouldCache || mCurrentBlockInStream.Source() == BlockInStreamSource.LOCAL) {
+    // do not create cache stream when the block is in local worker or from UFS
+    if (!mShouldCache || mCurrentBlockInStream.Source() == BlockInStreamSource.LOCAL
+        || mCurrentBlockInStream.Source() == BlockInStreamSource.UFS) {
```

Comment:
Simplify this to `mShouldCache && BlockInStreamSource.REMOTE`?

Reply:
done